documentation/error-handling.md

```diff
@@ -0,0 +1,64 @@
+# Error handling
+
+UI extensions support standard web errors. You can add `unhandledrejection` listener for unhandled promise rejections or `error` listener for other exceptions like Javascript runtime error or a resource fails to load.
+
+```javascript
+// For unhandled promise rejections
+self.addEventListener('unhandledrejection', (error) => {
+  console.warn('event unhandledrejection', error);
+});
+
+// For other exceptions
+self.addEventListener('error', (error) => {
+  console.warn('event error', error);
+});
+```
+
+## Reporting errors
+
+It's possible to use 3rd party libaries for reporting errors like Bugsnag or Sentry. However, it might require some extra configs or setup depending on which libraries you are using because UI extensions are run inside WebWorker.
+
+### Bugsnag
+
+Follow [Bugsnag documentation](https://docs.bugsnag.com/platforms/javascript/) to install it to your extension. If you use [CDN version](https://docs.bugsnag.com/platforms/javascript/cdn-guide/), you need to add polyfill code below as Bugsnag tries to access some variables that are not available in WebWorker.
+
+```javascript
+/**
+ * CDN polyfill
+ */
+(() => {
+  const document = {
+    documentElement: {
+      outerHTML: '',
+      createElement: () => ({}),
+      clientWidth: 0,
+      clientHeight: 0,
+    },
+    addEventListener: () => {},
+  };
+  const history = {};
+  const window = self;
+  self.window = window;
+  window.document = document;
+  window.history = history;
+})();
+```
```

Comment:
I know Chris mentioned it earlier but, yeah, this is a bug. If you get a chance, please file an issue for Bugsnag and link to it here (if possible) so that someone can see if it's fixed. 

This polyfill code seems fragile since a future version of Bugsnag could access different DOM properties.

Reply:
Did we file an issue for this?

Reply:
Also, do we think it's obvious enough where this code would go in an extension file? We don't show any context of what file/ where in a file we are adding this code, but not sure if we need to...